rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -326,7 +343,15 @@ void SequentialWriter::switch_to_next_storage()
   storage_options_.uri = format_storage_uri(
     base_folder_,
     metadata_.relative_file_paths.size());
+  
+  // when the storage_ is closed, rename the file to remove ".inactive" suffix
+  std::string active_path = storage_->get_relative_file_path();
+  std::string final_path = remove_active_from_filename(active_path);
+
   storage_ = storage_factory_->open_read_write(storage_options_);
+  
+  std::filesystem::rename(active_path, final_path);
```

Comment:
This could fail and generate the exception, but that is intention here. even though `std::filesystem::filesystem_error` is raised and caught here, there is pretty much nothing we can do here. so just raising exception upper layer would be fine.